pkg/controller/direct/secretmanager/secret_controller.go

```diff
@@ -122,6 +122,18 @@ func normalizeExternal(ctx context.Context, reader client.Reader, src client.Obj
 				secret.Spec.Replication.LegacyAutomatic.CustomerManagedEncryption.KmsKeyRef = kmsKeyRef
 			}
 		}
+		if secret.Spec.Replication.UserManaged != nil {
+			for _, r := range secret.Spec.Replication.UserManaged.Replicas {
+				if r.CustomerManagedEncryption != nil {
+					kmsKeyRef := r.CustomerManagedEncryption.KmsKeyRef
```

Comment:
I previously missed this field because I thought it is another field `spec.customerManagedEncryption`, which is not supported yet in the current released beta API.

Reply:
Do we have a fuzz test :-)

Reply:
PR on the way.